pkg/generate/app/cmd/newapp.go

```diff
@@ -418,6 +418,19 @@ func (c *AppConfig) buildPipelines(components app.ComponentReferences, environme
 						return nil, err
 					}
 				}
+				// Append any exposed ports from Dockerfile to input image
```

Comment:
An observation about the code.  It's really difficult to know which parts of the objects are supposed to be valid at any given point in time.  The `ComponentReference` holds a `ComponentInput` which could contain this information as early as the `componentsForRepos` step, but the `ComponentInput` gets stomped again during the `Resolve` step.  

Having this change here means that even after `Resolve` is finished, the `ComponentInput` is incomplete.  I *think* the `ComponentInput` is getting mutated here via  series of navigated references, but I'm not completely certain.  If so, it feels a little weird to have something called `buildPipelines` change an input object.